### DISCUSSION TITLE
JWS / Proof in createVerifiableCredential

### DISCUSSION THREAD

**Discussion by RobertoSnap** [decentralized-identity/veramo#571] [Q&A] [answered]
I want to make a proof that issuer X signs data {name: "Old Nordmann"} for subject Y.

I dont see how i should / can connect the VC and JWT? Or how i crate the proof in the agent.createVerifiableCredential() 

Here is my test https://github.com/BROKLab/auth-bot/blob/veramo/src/network/veramo.service.ts#L95

Could i get some pointers to how i should construct this to get the proof I am after?

**Comment by mirceanis** [comment, marked as answer]
`createVerifiableCredential` returns a Verifiable Credential object with a structure mimicking the ones described in the [W3C VC data model](https://www.w3.org/TR/vc-data-model/) for convenience.

When using `proofFormat: 'jwt'`, veramo creates a synthetic `JwtProof2020` proof property on the returned credential.
This proof type has a `jwt` field that contains the actual credential in [JWT encoding](https://www.w3.org/TR/vc-data-model/#jwt-encoding).

You can try something like this:
```ts
const issuerX = await agent.didManagerGet({ did: 'did:example:issuerX' })

const credential = {
  issuer: { id: issuerX.did }, // issuer DID MUST be an identifier managed by this agent ()
  credentialSubject: { 
    id: 'did:example:subjectY', 
    name: 'Old Nordmann' 
  },
}
const vc = await agent.createVerifiableCredential({
  credential,
  proofFormat: 'jwt',
})

console.log(JSON.stringify(vc, null, 2))
/*
{
  issuer: { id: 'did:example:issuerX' },
  credentialSubject: {
    name: 'Old Nordmann',
    id: 'did:example:subjectY'
  },
  type: [ 'VerifiableCredential' ],
  issuanceDate: '2021-06-15T07:27:01.000Z',
  proof: {
    type: 'JwtProof2020',
    jwt: 'eyJhbGci.....OCbMK8U1p6B0KzVA'
  },
  '@context': [ 'https://www.w3.org/2018/credentials/v1' ]
}
*/
```

Please note that `JwtProof2020` is not a registered proof type. It is only used for convenience, to be able to simplify the data model when working with JWT credential format.

I hope this helps
